### DISCUSSION TITLE
Applying Sketch color variables

### DISCUSSION THREAD

**Discussion by wimmulder** [ingrammicro/puzzle-tokens#44] [Q&A] [answered]
I am building a design system for our company and have implemented a bunch of text styles using Puzzle Tokens which has simplified keeping all text styles up to date a lot. Next thing I want to do is have Puzzle Tokens generate the layer styles that I am using. However, at the moment I am using the built in Sketch color variables. Is there a way to keep these applies to the layers when generating layer styles with Puzzle Tokens? That way I would keep the ability to change a color variable in Sketch and have it update everywhere. 
![Screenshot 2021-09-01 at 14 22 17](https://user-images.githubusercontent.com/553826/131670368-2ffee724-73d5-4138-9caa-1e29c65b60e0.png)
![Screenshot 2021-09-01 at 14 23 03](https://user-images.githubusercontent.com/553826/131670448-c097f35d-04c0-424c-8b4b-a3a162bbd238.png)

**Comment by wimmulder** [comment, marked as answer]
I think I found my own answer already, providing the variable name in double quotes seems to do the trick. Didn't see that in the docs before

**Comment by MaxBazarov** [comment]
Hi.
Actually the main page (https://github.com/ingrammicro/puzzle-tokens)
contains all required information.
But currently you need to search for "color var" text to find it.
Probably I need to highlight it  somehow.

Thanks for feedback!

ср, 1 сент. 2021 г. в 15:34, Wim Mulder ***@***.***>:

**Comment by wimmulder** [comment]
@MaxBazarov I've now created some layer styles based on the color variables defined in the Sketch file. But then I try to run the colors through a mix function as shown in this article (https://medium.com/sketch-app-sources/bringing-styles-preprocessing-to-sketch-9cdf0d0c49bd) :


```
@function tint($color, $value) {
    @return mix(#ffffff, $color, $value);
}

.Neutrals {
  @for $index from 1 through 7 {
    .Neutral-#{$index} {
      background-color: tint(map.get($colors, $key), 45% + ($index)*50%/7);
    } 
  }
}
```


This results in an error: "$color2: "Brand" is not a color."

When I fill this variable with a HEX color code it does calculate the color properly. Is there a way to run the color variable that is taken out of Sketch through the mix function in my scss file?

This is my list of variables:
$colors: (
    Brand:               "Brand",
    Primary:             "Primary", 
    Secondary:           "Secondary",
    Black:               "Black",
    White:               "White",
    Danger:              "Danger",
    Warning:             "Warning",
    Success:             "Success",
    Highlight:           "Highlight" 
);

This works fine for text styles and applies the correct color variables but seems not to work when running it through the function.

**Reply by MaxBazarov** [reply]
> @MaxBazarov I've now created some layer styles based on the color variables defined in the Sketch file. But then I try to run the colors through a mix function as shown in this article (https://medium.com/sketch-app-sources/bringing-styles-preprocessing-to-sketch-9cdf0d0c49bd) :
> 
> ```
> @function tint($color, $value) {
>     @return mix(#ffffff, $color, $value);
> }
> 
> .Neutrals {
>   @for $index from 1 through 7 {
>     .Neutral-#{$index} {
>       background-color: tint(map.get($colors, $key), 45% + ($index)*50%/7);
>     } 
>   }
> }
> ```
> 
> This results in an error: "$color2: "Brand" is not a color."
> 
> When I fill this variable with a HEX color code it does calculate the color properly. Is there a way to run the color variable that is taken out of Sketch through the mix function in my scss file?
> 
> This is my list of variables:
> $colors: (
> Brand: "Brand",
> Primary: "Primary",
> Secondary: "Secondary",
> Black: "Black",
> White: "White",
> Danger: "Danger",
> Warning: "Warning",
> Success: "Success",
> Highlight: "Highlight"
> );
> 
> This works fine for text styles and applies the correct color variables but seems not to work when running it through the function.

It looks like you are trying to use color variable value in SASS file, but it can't work in this way. Because SASS translator have no access to Sketch color variable values.  Do you understand me?
